src/slowmokit/methods/preprocessing/one_hot_encoder.cpp

```diff
@@ -0,0 +1,29 @@
+#include "one_hot_encoder.hpp"
+
+template<class T>
+
+// Function to one hot encode a vector of integers
+std::vector<std::vector<int>> one_hot_encoder(const std::vector<int> &data, int n_classes) {
```

Comment:
The data column might contain strings so better use template T. Remember we do one hot encoding for categorical variables

Reply:
done